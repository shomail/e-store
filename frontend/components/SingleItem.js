import React, { Component } from 'react';
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import Error from './ErrorMessage';

const SINGLE_ITEM_QUERY = gql`
  query SINGLE_ITEM_QUERY($id: ID!) {
    item(where: { id: $id }) {
      id
      title
      description
      largeImage
    }
  }
`;

class SingleItem extends Component {
  render() {
    const { id } = this.props;
    return (
      <Query query={SINGLE_ITEM_QUERY} variables={{ id }}>
        {({ error, loading, data }) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading!</p>;
          if (!data.item) return <p>No item found for {id}!</p>;
          return <p>Single Item Component - {id}</p>;
        }}
      </Query>
    );
  }
}

export default SingleItem;
