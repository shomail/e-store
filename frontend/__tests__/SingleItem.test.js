import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import SingleItem, { SINGLE_ITEM_QUERY } from '../components/SingleItem';
import { fakeItem } from '../lib/testUtils';

describe('<SingleItem />', () => {
  it('Renders with proper data', async () => {
    const mocks = [
      {
        // when someone makes a request with this query and variable
        request: { query: SINGLE_ITEM_QUERY, variables: { id: '123' } },
        // return this fake data
        result: {
          data: {
            item: fakeItem(),
          },
        },
      },
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );
    expect(wrapper.text()).toContain('Loading!');
  });
});
