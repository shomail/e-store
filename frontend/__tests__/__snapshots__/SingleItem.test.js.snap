// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem /> Renders with proper data 1`] = `
<h2>
  Viewing 
  dogs are best
</h2>
`;

exports[`<SingleItem /> Renders with proper data 2`] = `
<img
  alt="dogs are best"
  src="dog.jpg"
/>
`;

exports[`<SingleItem /> Renders with proper data 3`] = `
<p>
  dogs
</p>
`;
